### ISSUE TITLE
Exceptions ignored when using send_async with exponential_retry tasks

### ISSUE THREAD

**Issue by curzonj** [aws/aws-flow-ruby#100]
Due to

https://github.com/aws/aws-flow-ruby/blob/master/aws-flow/lib/aws/decider/generic_client.rb#L100
https://github.com/aws/aws-flow-ruby/blob/master/aws-flow/lib/aws/decider/activity.rb#L107
https://github.com/aws/aws-flow-ruby/blob/master/aws-flow/lib/aws/decider/generic_client.rb#L149

If you try to use send_async on a task configured for exponential_retry, if it still fails after retry, the error will be swallowed and your error_handler won't see any exceptions. This doesn't happen if you call the activity synchronously.

**Comment by mjsteger** [comment]
Yeah, definitely seeing the same behavior. I see that you have a fix on your branch, which looks like the right approach. I'm going to cherrypick that and add a test

**Comment by mustafashabib** [comment]
Did this get merged?

**Comment by mjsteger** [comment]
Not yet, just created a PR for it. Not sure how long it'll take for a response, so you may want to consider going off my git branch/forking it yourself. Only thing to watch for is that while I'm pretty sure that my changes are correct, I can't run the integ tests very much without incurring some expenses, so the code is only validated with unit tests for now.
